thrift/annotation/rust.thrift

```diff
@@ -65,3 +65,10 @@ struct Adapter {
   // will use `fbthrift_adapters::DurationSecondsAdapter<i64>` as the adapter.
   1: string name;
 } (thrift.uri = "facebook.com/thrift/annotation/rust/Adapter")
+
+@scope.Exception
+@scope.Field
+@scope.Struct
+struct Attributes{
```

Comment:
Can you change this to `@scope.Field`? If you want to apply a macro to a Thrift struct, you should use the `rust.derive = "..."` unstructured annotation (exists already) or the `@rust.Derives { derive = [...] }` structured annotation (will be putting up a PR for this shortly). 

Attribute macros allow you to read/modify the generated struct definition, which opens the door for a number of safety/compatibility issues. Derive macros allow only read-access to the struct definition (and subsequently enforce append-only behavior for macro generated code), which is much safer and should be the only allowed behavior for Rust Thrift.